### DISCUSSION TITLE
Is Workerman appropriate for database queries?

### DISCUSSION THREAD

**Discussion by ghost** [walkor/Workerman#1045] [Q&A] [answered]
Hello there,

At [ChesslaBlab](https://github.com/chesslablab), we have on the one hand an asynchronous [PHP Chess Server](https://github.com/chesslablab/chess-server) for chess commands that are intended to run very quickly in real-time. On the other, we have a [PHP Chess API](https://github.com/chesslablab/chess-api) intended to run database queries that may take a few seconds to execute. Up until now we've made a distinction between real-time and non-real time operations.

Real-time operations are implemented in an async chess server while non-real time operations are implemented in a chess api. The reasoning behind is, this way users who need to run operations in real-time won't be affected by those who need to run operations that may take some time to execute.

However, now we're considering to remove the API and implement all operations in the form of chess commands.

So, here's the question. How would users playing chess online be affected by those who are querying the database?

Thanks for the help, and keep up with the great work you're doing!

**Comment by ghost** [comment]
See:

- https://github.com/chesslablab/chess-server/issues/311

**Comment by joanhey** [comment, marked as answer]
Absolutely it is faster and with less latency to use database queries !! 

A simple query benchmark req/s :

Workerman with postgresql
![image](https://github.com/user-attachments/assets/3b827637-d8f2-47a8-a9ee-3773d62ed642)

PHP raw with postgresql
![image](https://github.com/user-attachments/assets/cfa2b26d-3a88-4642-a451-7f08a6525943)

Example using Symfony app (with Adapterman) single query with full ORM:

![image](https://github.com/user-attachments/assets/a2f9e6c3-55ee-4f30-96f4-187ce5bba94e)

And if you only need to save to the database, you can make it after send the response.

**Comment by ghost** [comment]
See: 

- https://github.com/chesslablab/chess-server/issues/354
